lib/Doctrine/DBAL/Platforms/MySQL578Platform.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Platforms;
+
+/**
+ * Provides the behavior, features and SQL dialect of the MySQL >=5.7.8 database platform.
+ * Extends for json support >=5.7.8 version
+ *
+ * @author İsmail BASKIN <ismailbaskin1@gmail.com>
+ * @link   www.doctrine-project.org
+ * @since  2.5
+ */
+class MySQL578Platform extends MySQL57Platform
```

Comment:
If we introduce a new platform, we also need to introduce a new reserved keywords class. See `MySQL57Platform` for reference. I had the time to get a diff of keywords between 5.7.0 and 5.7.8 (according to the [documentation](http://dev.mysql.com/doc/refman/5.7/en/keywords.html):

*New reserved keywords*
- `GENERATED`
- `OPTIMIZER_COSTS`
- `STORED`
- `VIRTUAL`

*Removed reserved keywords*
- `NONBLOCKING`

The new keywords class can simply extend `MySQL57Keywords` class and then you can `array_merge` and `array_diff` the 5.7.0 keywords from parent.